src/Ampersand/PatchHelper/Helper/Magento2Instance.php

```diff
@@ -119,6 +102,107 @@ public function __construct(string $path)
+                    break;
+            }
+        }
+        unset($theme);
+
+        try {
+            /** @var \Magento\Store\Model\StoreManagerInterface $storeManager */
+            $storeManager = $this->objectManager->get(\Magento\Store\Model\StoreManagerInterface::class);
+            /** @var \Magento\Theme\Model\View\Design $design */
+            $design = $this->objectManager->get(\Magento\Theme\Model\View\Design::class);
+
+            $usedFrontendThemes = [];
+            foreach ($storeManager->getStores(false) as $store) {
+                $themeCode = $design->getConfigurationDesignTheme(
+                    'frontend',
+                    [
+                        'store' => $store->getId()
+                    ]
+                );
+
+                if (isset($allFrontendThemes[$themeCode])) {
+                    $usedFrontendThemes[$themeCode] = $allFrontendThemes[$themeCode];
+                }
+            }
+            unset($themeCode, $store);
+
+            $frontendThemesToScan = [];
+            foreach ($usedFrontendThemes as $usedFrontendTheme) {
+                while ($usedFrontendTheme) {
+                    if (!str_starts_with($usedFrontendTheme->getCode(), 'Magento/')) {
+                        $frontendThemesToScan[$usedFrontendTheme->getCode()] = $usedFrontendTheme;
+                    }
+                    $usedFrontendTheme = $usedFrontendTheme->getParentTheme();
+                }
+            }
+            unset($usedFrontendThemes, $usedFrontendTheme);
+
+            foreach ($allFrontendThemes as $frontendTheme) {
+                if (!isset($frontendThemesToScan[$frontendTheme->getCode()])) {
+                    $themesToIgnore[$frontendTheme->getCode()] = $frontendTheme;
+                }
+            }
+            unset($frontendTheme);
+
+            $this->customFrontendThemes = $frontendThemesToScan;
+        } catch (\Throwable $throwable) {
+            // We likely couldn't grab the theme information from database, default to using all custom themes
+            $this->customFrontendThemes =  [];
+            foreach ($themeList as $theme) {
+                // ignore Magento themes
+                if (strpos($theme->getCode(), 'Magento/') === 0) {
+                    continue;
+                }
+                switch ($theme->getArea()) {
+                    case Area::AREA_FRONTEND:
+                        $this->customFrontendThemes[] = $theme;
+                        break;
+                }
+            }
+        }
+
+        /**
+         * TODO hyva_theme_fallback/general/theme_full_path
+         *
+         * @link https://docs.hyva.io/hyva-themes/luma-theme-fallback/index.html
+         *
+         * check if we have a usable database connection to grab the value of this config, and
+         * include that custom frontend theme from the nonHyvaThemes list
+         *
+         * Will this always be a custom theme or can people put fallbacks to Luma ?
```

Comment:
The fallback theme can be any theme - during development it's usually native Luma, in practice it always is a regular customized Luma theme.

Reply:
@Vinai is this usually configured / enabled by default? Or is it something that most people do not use?

@norgeindian with this setting configured we in effect have Luma and all the traditional magento theming within Hyva anyway, which means that we still have to scan for most of the system which means the current implementation does it more or less okay? 

Unless i'm missing something? 

It would be helpful if you could provide example output for overrides which you think are valid, and ones that you believe should be ignored.

The current work in this PR will ensure (when theming and store information is available, either from a `config.php` setting or from the DB) that we only ever scan for overrides in actively assigned custom themes, which would go some way to ignoring Luma etc if you had a pure hyva store with no fallback configured.

Reply:
@convenient , exactly. As soon as this setting is set, it could be any theme. Like @Vinai  already wrote, it is usually a luma based custom theme, at least in our cases.
I will try to test this approach soon and give you feedback.

Reply:
@amenk 

https://github.com/AmpersandHQ/ampersand-magento2-upgrade-patch-helper/issues/75#issuecomment-1330725137

Theme fallback still a work in progress.

What kind of value do you have in the configuration? Is it a theme code that we can look up the value from core config data, and then grab the theme from the array that we manipulated this morning and include that in the valid themes list?

Reply:
Yes, that is true.  
Hyvä has this concept of "compatibility modules".
A compatibility module provides the Hyvä templates and layout XML for a given "original module" that was made for Luma.

Hyvä compatibility modules have a feature that allows them to "register" themselves with as a compatibility module for one or more original modules in `etc/frontend/di.xml`, and the they can override templates from a original module by simply creating them in their own `view/frontend/templates` folder structure with the same name.  
The benefit is that less Layout XML boilerplate code is needed.

This feature is provided by the [hyva-themes/magento2-compat-module-fallback](https://gitlab.hyva.io/hyva-themes/magento2-compat-module-fallback) module, which is added as a dependency to compatibility modules using this feature.  
More information about the `di.xml` data structure can be found in the README file of that module.  

Technically, it injects the compatibility module view directory into the theme fallback lookup array during runtime whenever a template for a given module.

The class `Hyva\CompatModuleFallback\Model\`CompatModuleRegistry` has methods that can be used to find compatibility modules for a given original module. Maybe they are useful if you want to include this in the upgrade helper.
In particular I'm thinking of

* `CompatModuleRegistry::getCompatModulesFor(string $originalModule): array`
* `CompatModuleRegistry::getCompatModules(): array`
* `CompatModuleRegistry::getOrigModules(): array`

Reply:
@convenient how would we integrate that ?

Reply:
Maybe something like
- if I'm scanning for `vendor/magento/module-catalog/view/base/templates/product/price/tier_prices.phtml`
- if that theme file path also exists in a theme starting with `Hyva/` 
    -  skip this scan, because we know hyva is the source of it.
    
That way any filepaths that aren't defined in Hyva would continue to scan as normal, and this case would be covered.
    
🤔

Reply:
The fallbacks are ordered. So if there is `tier_prices.phtml` in Hyvä it is very likely that the template was copied from there. In case the template is not there, the next fallback could be used.
Should also work for the per-module fallbacks.

Reply:
Okay, it *is* (of-course) already using the [fallback resolution](https://github.com/AmpersandHQ/ampersand-magento2-upgrade-patch-helper/blob/4e464bdc2b4a31e5b49cb149a2fe5ca1c9b2c051/src/Ampersand/PatchHelper/Checks/FrontendFilePhtml.php#L57) - if would guess, that this also should apply in Hyvä and no special coding would me needed?

Reply:
Hmm @amenk Did 

```
| WARN | Override (phtml/js/html) | vendor/magento/module-catalog/view/base/templates/product/price/tier_prices.phtml | app/design/frontend/OurCustomer/Hyvaaaaa/Magento_Catalog/templates/product/price/tier_prices.phtml |
```

also get reported for the other themes? For the base hyva themes as well as your own customisation?

Reply:
@convenient On the left hand side of the report, I did not get a single hyva-phtml override.

Reply:
For that file on the left hand side, how many right hand sides did you get a hit of?

Was Hyva upgraded as part of your recent run? You'd only expect to see files on the left that have changed.

Reply:
I got one for each theme (we have 2 Hyvä based subthemes)
Ah, and yes, I did a Hyva upgrade as part of my run.
But it don't know if that file template was also changed in Hyvä, maybe that is my misunderstanding.

Reply:
>But it don't know if that file template was also changed in Hyvä, maybe that is my misunderstanding.

If you still have the `vendor.patch` file you should be able to verify that.

Okay that's all good information to know thanks

Reply:
Yes, in the vendor.patch I for example see changes like this:

```vendor/hyva-themes/magento2-default-theme/Magento_Customer/templates/address/edit.phtml``` in the output I see this:

```
| WARN  | Override (phtml/js/html) | vendor/magento/module-customer/view/frontend/templates/address/edit.phtml                                                   | app/design/frontend/OurCustomer/Hyvaaaaa/Magento_Customer/templates/address/edit.phtml                                    |
| WARN  | Override (phtml/js/html) | vendor/magento/module-customer/view/frontend/templates/address/edit.phtml                                                   | vendor/hyva-themes/magento2-default-theme/Magento_Customer/templates/address/edit.phtml                               |
```

I somehow would have expected 

```
| WARN  | Override (phtml/js/html) | vendor/hyva-themes/magento2-default-theme/Magento_Customer/templates/address                                               | app/design/frontend/OurCustomer/Hyvaaaaa/Magento_Customer/templates/address/edit.phtml                                    |
| WARN  | Override (phtml/js/html) | vendor/magento/module-customer/view/frontend/templates/address/edit.phtml                                                   | vendor/hyva-themes/magento2-default-theme/Magento_Customer/templates/address/edit.phtml                               |
```

To show differences on all levels of the fallback structure, but not sure if that makes sense and if it is supposed to work like that :)

Reply:
Thanks for the details. Don't have an actual Hyva instance myself but will give this a proper think and see what I can do to improve things

Reply:
This seems to agree with what @norgeindian said in a main PR comment that Hyva is missing on the left hand side for some reason. Will have to see how that compares to my test case

Reply:
@amenk You said you would expect

```
| WARN  | Override (phtml/js/html) | vendor/hyva-themes/magento2-default-theme/Magento_Customer/templates/address/edit.phtml                                               | app/design/frontend/OurCustomer/Hyvaaaaa/Magento_Customer/templates/address/edit.phtml                                    |
| WARN  | Override (phtml/js/html) | vendor/magento/module-customer/view/frontend/templates/address/edit.phtml                                                   | vendor/hyva-themes/magento2-default-theme/Magento_Customer/templates/address/edit.phtml                               |
```

Would you really expect to be notified that `vendor/magento/module-customer/view/frontend/templates/address/edit.phtml` has been updated and that `vendor/hyva-themes/magento2-default-theme/Magento_Customer/templates/address/edit.phtml` overrides it? I thought the point of hyva was that it was a complete replacement and that these could be ignored in this scenario.

Reply:
You a right, the second line is optional.. the first is the important one.

Reply:
@amenk you and @norgeindian were correct in that the custom themes, or even magento themes, weren't being reported on the left hand side properly.

I've actually captured this in a test case now, and see my test output like 

```
| WARN  | Override (phtml/js/html) | vendor/ampersand/upgrade-patch-helper-test-module/src/theme/Magento_Checkout/templates/cart/form.phtml | app/design/frontend/Ampersand/theme/Magento_Checkout/templates/cart/form.phtml                 |
| WARN  | Override (phtml/js/html) | vendor/magento/module-sales/view/frontend/email/order_new.html                                         | vendor/ampersand/upgrade-patch-helper-test-module/src/theme/Magento_Sales/email/order_new.html |
| WARN  | Override (phtml/js/html) | vendor/magento/theme-frontend-luma/Magento_Sales/email/order_new.html                                  | vendor/ampersand/upgrade-patch-helper-test-module/src/theme/Magento_Sales/email/order_new.html |
```

This verifies that custom theme changes will be reported on the left, as well as changes to core magento themes being reported.

I've also considered a "less is more" approach to developing this, that's lower risk for accidentally excluding themes that people may have grown used to reporting on.

This approach scans all the themes as the tool has always done, but (hopefully) if we're dealing with Hyva themes it will detect and handle the above case, where hyva "owns" the template path.

The new Hyva "exclusion" logic can be seen in `src/Ampersand/PatchHelper/Checks/FrontendFilePhtml.php`, I've only hacked it up into that class and put TODOs into the other relevant places until it's verified that that approach makes sense, or works.

@amenk fancy taking this for a test drive? Hopefully you should see 
1. More phtml files reported on the left side
2. Cases like `vendor/magento/module-customer/view/frontend/templates/address/edit.phtml` not reporting hyva base themes (because they own that file and not really an override)

Reply:
I worked a bit with it now, and I cannot spot any issues - works wonderfully I would say.
Even the compat-module-fallback seems to work.
Thank you very much :-)

Reply:
Woah @amenk !  I'm not saying I'm surprised, but maybe a little. Would you mind sharing some of the real kind of output?  If you have any examples of email template or layout xml changes that would be helpful but also curious to see how the phtml files are reported if they are as we want.

Currently I have only targeted `.phtml` files to remove redundant output.

The outstanding frontend checks are

email templates `.html`
```
| WARN  | Override (phtml/js/html) | vendor/magento/module-sales/view/frontend/email/order_new.html                                                                     | vendor/ampersand/upgrade-patch-helper-test-module/src/theme/Magento_Sales/email/order_new.html                                     |
```

web templates `.html`  
```
| WARN  | Override (phtml/js/html) | vendor/magento/module-checkout/view/frontend/web/template/summary/item/details.html                                                | app/design/frontend/Ampersand/theme/Magento_Checkout/web/template/summary/item/details.html                                        |
```

layout xml overrides /extensions 
```
| WARN  | Override (phtml/js/html) | vendor/magento/module-sales/view/frontend/layout/sales_order_print.xml                                                             | app/design/frontend/Ampersand/theme/Magento_Sales/layout/sales_order_print.xml
```

The web templates `.html` would mostly be knockout templates IIRC, which are probably not relevant in hyva so nothing to do for those @Vinai ?

Reply:
Here: https://github.com/AmpersandHQ/ampersand-magento2-upgrade-patch-helper/pull/80#issuecomment-1335443686

Reply:
Thats really cool @amenk 

I think my filtering is not working correctly as we don't really need to see this in the report I think

```
| WARN  | Override (phtml/js/html) | vendor/magento/module-sales/view/frontend/templates/order/items/renderer/default.phtml                                       | vendor/hyva-themes/magento2-default-theme/Magento_Sales/templates/order/items/renderer/default.phtml                    |
```

If you have the time and could run through xebug to see why `src/Ampersand/PatchHelper/Checks/FrontendFilePhtml.php` is not hitting the "Handle hyva themes" `continue;` function? no probs if not

Reply:
@amenk is it this? is this my mistake?

```diff
diff --git a/src/Ampersand/PatchHelper/Checks/FrontendFilePhtml.php b/src/Ampersand/PatchHelper/Checks/FrontendFilePhtml.php
index 95cf27a..88824f4 100644
--- a/src/Ampersand/PatchHelper/Checks/FrontendFilePhtml.php
+++ b/src/Ampersand/PatchHelper/Checks/FrontendFilePhtml.php
@@ -71,7 +71,7 @@ class FrontendFilePhtml extends AbstractCheck
                             // We are investigating a vendor/magento template change that exists in a hyva base theme
                             // This suggests that hyva is the originator of this template, not magento
                             // We should only report this vendor/magento in non hyva based themes
-                            continue;
+                            continue 2;
                         }
                     } catch (\InvalidArgumentException $exception) {
                         // lookup failed, not in this hyva theme
```

Will that prevent the overreporting of theme modifications we dont really care about?

Reply:
so I am running

`analyse projectfolder --filter renderer/default.phtml`

It does hit src/Ampersand/PatchHelper/Checks/FrontendFilePhtml.php:74 continue, but not always...

So I added some debug code:

```
--- a/src/Ampersand/PatchHelper/Checks/FrontendFilePhtml.php
+++ b/src/Ampersand/PatchHelper/Checks/FrontendFilePhtml.php
@@ -60,6 +60,9 @@ class FrontendFilePhtml extends AbstractCheck
              *
              * TODO repeat / extract / etc similar logic into other necessary Checks
              */
+
+            echo $this->patchEntry->getPath() . ' ';
+
             if (
                 str_starts_with($this->patchEntry->getPath(), 'vendor/magento/') &&
                 isset($hyvaThemes[$theme->getCode()])
@@ -71,14 +74,18 @@ class FrontendFilePhtml extends AbstractCheck
                             // We are investigating a vendor/magento template change that exists in a hyva base theme
                             // This suggests that hyva is the originator of this template, not magento
                             // We should only report this vendor/magento in non hyva based themes
+                            echo "continue ";
                             continue;
                         }
                     } catch (\InvalidArgumentException $exception) {
+                        echo "lookup failed ";
                         // lookup failed, not in this hyva theme
                     }
                 }
             }
 
+            echo "after if" . PHP_EOL;
+
```

```
--- a/src/Ampersand/PatchHelper/Checks/FrontendFilePhtml.php
+++ b/src/Ampersand/PatchHelper/Checks/FrontendFilePhtml.php
@@ -60,6 +60,9 @@ class FrontendFilePhtml extends AbstractCheck
              *
              * TODO repeat / extract / etc similar logic into other necessary Checks
              */
+
+            echo $this->patchEntry->getPath() . ' ';
+
             if (
                 str_starts_with($this->patchEntry->getPath(), 'vendor/magento/') &&
                 isset($hyvaThemes[$theme->getCode()])
@@ -71,14 +74,18 @@ class FrontendFilePhtml extends AbstractCheck
                             // We are investigating a vendor/magento template change that exists in a hyva base theme
                             // This suggests that hyva is the originator of this template, not magento
                             // We should only report this vendor/magento in non hyva based themes
+                            echo "continue ";
                             continue;
                         }
                     } catch (\InvalidArgumentException $exception) {
+                        echo "lookup failed ";
                         // lookup failed, not in this hyva theme
                     }
                 }
             }
 
+            echo "after if" . PHP_EOL;
+
vendor/hyva-themes/magento2-default-theme/Magento_Sales/templates/order/items/renderer/default.phtml after if
vendor/hyva-themes/magento2-default-theme/Magento_Sales/templates/order/items/renderer/default.phtml after if
vendor/hyva-themes/magento2-default-theme/Magento_Sales/templates/order/items/renderer/default.phtml after if
vendor/magento/module-sales/view/frontend/templates/order/items/renderer/default.phtml continue continue after if
vendor/magento/module-sales/view/frontend/templates/order/items/renderer/default.phtml continue continue after if
vendor/magento/module-sales/view/frontend/templates/order/items/renderer/default.phtml continue continue after if
+-------+--------------------------+----------------------------------------------------------------------------------------+------------------------------------------------------------------------------------------------------+
| Level | Type                     | File                                                                                   | To Check                                                                                             |
+-------+--------------------------+----------------------------------------------------------------------------------------+------------------------------------------------------------------------------------------------------+
| WARN  | Override (phtml/js/html) | vendor/magento/module-sales/view/frontend/templates/order/items/renderer/default.phtml | vendor/hyva-themes/magento2-default-theme/Magento_Sales/templates/order/items/renderer/default.phtml |
+-------+--------------------------+----------------------------------------------------------------------------------------+------------------------------------------------------------------------------------------------------+
WARN count: 1
```

What does this tell us?

Reply:
>It does hit src/Ampersand/PatchHelper/Checks/FrontendFilePhtml.php:74 continue, but not always...

That's good! 

@amenk Please see my above comment, i think i meant to do `continue 2;` 🤞

Reply:
> @amenk is it this? is this my mistake?

> Will that prevent the overreporting of theme modifications we dont really care about?

I guess so, seems to work.

At least that file is no longer in the output: https://github.com/AmpersandHQ/ampersand-magento2-upgrade-patch-helper/files/10142596/patch-helper-after-continue.txt

Reply:
The files might not be fully representative, as I already worked on porting the changes in the meantime...

Reply:
@amenk To me that's looking very good with one question.

This one is no longer reported, that is unexpected to me as the code change should only affect `vendor/magento` files on the left?
```
| WARN  | Override (phtml/js/html) | vendor/hyva-themes/magento2-default-theme/Magento_Customer/templates/account/authentication-popup.phtml                      | app/design/frontend/OurCustomer/Hyvaaaaa/Magento_Customer/templates/account/authentication-popup.phtml                      |
```

And all of these were removed from the report, which makes sense as they are `vendor/magento` templates but hyva is the actual source of them. I think it makes sense that you don't need to see these in the report, would you agree?

```
| WARN  | Override (phtml/js/html) | vendor/magento/module-bundle/view/base/templates/product/price/tier_prices.phtml                                             | vendor/hyva-themes/magento2-default-theme/Magento_Bundle/templates/product/price/tier_prices.phtml                      |
| WARN  | Override (phtml/js/html) | vendor/magento/module-catalog/view/base/templates/product/price/tier_prices.phtml                                            | app/design/frontend/OurCustomer/Hyvaaaaa/Magento_Catalog/templates/product/price/tier_prices.phtml                          |
| WARN  | Override (phtml/js/html) | vendor/magento/module-catalog/view/base/templates/product/price/tier_prices.phtml                                            | vendor/hyva-themes/magento2-default-theme/Magento_Catalog/templates/product/price/tier_prices.phtml                     |
| WARN  | Override (phtml/js/html) | vendor/magento/module-catalog/view/frontend/templates/navigation/left.phtml                                                  | vendor/hyva-themes/magento2-default-theme/Magento_Catalog/templates/navigation/left.phtml                               |
| WARN  | Override (phtml/js/html) | vendor/magento/module-catalog/view/frontend/templates/product/list/toolbar/amount.phtml                                      | app/design/frontend/OurCustomer/Hyvaaaaa/Magento_Catalog/templates/product/list/toolbar/amount.phtml                        |
| WARN  | Override (phtml/js/html) | vendor/magento/module-catalog/view/frontend/templates/product/list/toolbar/amount.phtml                                      | vendor/hyva-themes/magento2-default-theme/Magento_Catalog/templates/product/list/toolbar/amount.phtml                   |
| WARN  | Override (phtml/js/html) | vendor/magento/module-catalog/view/frontend/templates/product/list/toolbar/limiter.phtml                                     | app/design/frontend/OurCustomer/Hyvaaaaa/Magento_Catalog/templates/product/list/toolbar/limiter.phtml                       |
| WARN  | Override (phtml/js/html) | vendor/magento/module-catalog/view/frontend/templates/product/list/toolbar/limiter.phtml                                     | app/design/frontend/OurCustomer/TangaSports/Magento_Catalog/templates/product/list/toolbar/limiter.phtml                    |
| WARN  | Override (phtml/js/html) | vendor/magento/module-catalog/view/frontend/templates/product/list/toolbar/limiter.phtml                                     | vendor/hyva-themes/magento2-default-theme/Magento_Catalog/templates/product/list/toolbar/limiter.phtml                  |
| WARN  | Override (phtml/js/html) | vendor/magento/module-configurable-product/view/base/templates/product/price/tier_price.phtml                                | vendor/hyva-themes/magento2-default-theme/Magento_ConfigurableProduct/templates/product/price/tier_price.phtml          |
| WARN  | Override (phtml/js/html) | vendor/magento/module-customer/view/frontend/templates/account/authentication-popup.phtml                                    | app/design/frontend/OurCustomer/Hyvaaaaa/Magento_Customer/templates/account/authentication-popup.phtml                      |
| WARN  | Override (phtml/js/html) | vendor/magento/module-customer/view/frontend/templates/account/authentication-popup.phtml                                    | vendor/hyva-themes/magento2-default-theme/Magento_Customer/templates/account/authentication-popup.phtml                 |
| WARN  | Override (phtml/js/html) | vendor/magento/module-customer/view/frontend/templates/address/edit.phtml                                                    | app/design/frontend/OurCustomer/Hyvaaaaa/Magento_Customer/templates/address/edit.phtml                                      |
| WARN  | Override (phtml/js/html) | vendor/magento/module-customer/view/frontend/templates/address/edit.phtml                                                    | vendor/hyva-themes/magento2-default-theme/Magento_Customer/templates/address/edit.phtml                                 |
| WARN  | Override (phtml/js/html) | vendor/magento/module-customer/view/frontend/templates/form/forgotpassword.phtml                                             | app/design/frontend/OurCustomer/Hyvaaaaa/Magento_Customer/templates/form/forgotpassword.phtml                               |
| WARN  | Override (phtml/js/html) | vendor/magento/module-customer/view/frontend/templates/form/forgotpassword.phtml                                             | vendor/hyva-themes/magento2-default-theme/Magento_Customer/templates/form/forgotpassword.phtml                          |
| WARN  | Override (phtml/js/html) | vendor/magento/module-customer/view/frontend/templates/form/login.phtml                                                      | app/design/frontend/OurCustomer/Hyvaaaaa/Magento_Customer/templates/form/login.phtml                                        |
| WARN  | Override (phtml/js/html) | vendor/magento/module-customer/view/frontend/templates/form/login.phtml                                                      | app/design/frontend/OurCustomer/TangaSports/Magento_Customer/templates/form/login.phtml                                     |
| WARN  | Override (phtml/js/html) | vendor/magento/module-customer/view/frontend/templates/form/login.phtml                                                      | vendor/hyva-themes/magento2-default-theme/Magento_Customer/templates/form/login.phtml                                   |
| WARN  | Override (phtml/js/html) | vendor/magento/module-customer/view/frontend/templates/form/register.phtml                                                   | app/design/frontend/OurCustomer/Hyvaaaaa/Magento_Customer/templates/form/register.phtml                                     |
| WARN  | Override (phtml/js/html) | vendor/magento/module-customer/view/frontend/templates/form/register.phtml                                                   | app/design/frontend/OurCustomer/TangaSports/Magento_Customer/templates/form/register.phtml                                  |
| WARN  | Override (phtml/js/html) | vendor/magento/module-customer/view/frontend/templates/form/register.phtml                                                   | vendor/hyva-themes/magento2-default-theme/Magento_Customer/templates/form/register.phtml                                |
| WARN  | Override (phtml/js/html) | vendor/magento/module-layered-navigation/view/frontend/templates/layer/filter.phtml                                          | vendor/hyva-themes/magento2-default-theme/Magento_LayeredNavigation/templates/layer/filter.phtml                        |
| WARN  | Override (phtml/js/html) | vendor/magento/module-page-builder/view/frontend/templates/catalog/product/widget/content/carousel.phtml                     | vendor/hyva-themes/magento2-default-theme/Magento_PageBuilder/templates/catalog/product/widget/content/carousel.phtml   |
| WARN  | Override (phtml/js/html) | vendor/magento/module-sales/view/frontend/templates/order/items/renderer/default.phtml                                       | vendor/hyva-themes/magento2-default-theme/Magento_Sales/templates/order/items/renderer/default.phtml                    |
| WARN  | Override (phtml/js/html) | vendor/magento/module-swatches/view/frontend/templates/product/layered/renderer.phtml                                        | vendor/hyva-themes/magento2-default-theme/Magento_Swatches/templates/product/layered/renderer.phtml                     |
| WARN  | Override (phtml/js/html) | vendor/magento/module-theme/view/frontend/templates/html/pager.phtml                                                         | app/design/frontend/OurCustomer/Hyvaaaaa/Magento_Theme/templates/html/pager.phtml                                           |
| WARN  | Override (phtml/js/html) | vendor/magento/module-theme/view/frontend/templates/html/pager.phtml                                                         | vendor/hyva-themes/magento2-default-theme/Magento_Theme/templates/html/pager.phtml                                      |
```

Reply:
Simple, I deleted that file in the meantime :-))

Reply:
@amenk 🙌 

The only "outstading" tasks here form my perspective now are 
1. to decide what happens with the `.html` and `.xml` override detection.
2. tidy up the code, comments, and add some more tests

I'm thinking hyva probably doesnt do email, or knockout templates so i can probably ignore the `.html`. 

And for the `.xml` the changes there are less common, so it may do some over-reporting like `vendor/magento/some/theme/file.xml` being reported in hyva, but the effort for developing a similar filtering mechanism is a bit higher.

You have helped so much 👍 🥳

Reply:
Happy I could help. Thanks for this amazing tool.